proc.c

```diff
@@ -628,3 +631,59 @@ procdump(void)
       cprintf("\n");
   }
 }
+
+
+int kthread_create(void (*start_func)(), void* stack){
+    //TODO
+    return 0;
+}
+
+int kthread_id(){
+    return my_thread()->tid;
+}
+
+void kthread_exit(){
+    //struct proc *curproc = myproc();
+    struct thread * curr_thread = my_thread();
+    boolean lastthread = TRUE;
+    struct proc* proc = myproc();
+    acquire(&proc->proclock);
+    struct thread * t;
+    for (t = proc->threads; t < &proc->threads[NTHREADS]; t++){
+        if(t->t_state == T_RUNNABLE | t->t_state==T_RUNNING |t->t_state == T_SLEEPING) {
+            lastthread = FALSE;
+            break;
+        }
+
+    }
+    if(lastthread){
+        exit();
```

Comment:
I think we should handle the case where exit is called before kthread_exit is called.
but maybe it's best to wait until we implement thread_join, what do you think?